src/ott/initializers/linear/initializers.py

```diff
@@ -313,3 +332,108 @@ def body_fn(i: int, f: jnp.ndarray) -> jnp.ndarray:
     return f.at[i].set(new_f)
 
   return jax.lax.fori_loop(0, len(f), body_fn, f)
+
+
+@functools.partial(jax.jit, static_argnums=(3,))
+def subsample(
+    rng: jax.random.PRNGKey, array: jnp.ndarray, weight: jnp.ndarray,
+    subsample_n: int
+):
+  """Subsample point cloud to uniform weighted points.
+
+  Args:
+    rng: random number generator.
+    array: point cloud, array of size N x d.
+    weight: array of size N.
+    subsample_n: number of points to subsample.
+
+  Returns:
+    subsampled point cloud, array of size subsample_n x d.
+  """
+  N = array.shape[0]
+  indices = jax.random.choice(
+      key=rng, a=N, shape=(subsample_n,), replace=True, p=weight, axis=0
+  )
+  return array[indices]
+
+
+@jax.tree_util.register_pytree_node_class
+class SubsampleInitializer(DefaultInitializer):
+  """Subsample initializer :cite:`thornton2022rethinking:22`.
+
+  Subsample each point cloud, then compute closed from
+  Sinkhorn potential between subsampled approximations. Use this potential
+  to initialize Sinkhorn potentials/scalings.
+
+  Args:
+    subsample_n: number of points to subsample from each point cloud.
+  """
+
+  def __init__(
+      self,
+      subsample_n: int,
+  ):
+    super().__init__()
+
+    self.subsample_n = subsample_n
+
+  def solve_sub_problem(self, sub_x, sub_y, epsilon):
+    """Solve subproblem with  Sinkhorn.
+
+    Args:
+      sub_x: subsampled x, array of size subsample_n x d.
+      sub_y: subsampled y, array of size subsample_n x d.
+      epsilon: regularization parameter for subsampled point cloud.
+
+    Returns:
+      Sinkhorn output object.
+    """
+    from ott.solvers.linear import sinkhorn
+
+    # can extract cost matrix entries or recompute
+    sub_geom = pointcloud.PointCloud(x=sub_x, y=sub_y, epsilon=epsilon)
+    ot_prob = linear_problem.LinearProblem(sub_geom)
```

Comment:
Naive question: shouldn't this also inherit `tau_{a,b}` from the original problem? Maybe even `a/b`, although they are implicitly there because of the subsampling weights.

Reply:
Good question, let me think about it.